turbinia/evidence.py

```diff
@@ -304,6 +315,39 @@ def __str__(self):
   def __repr__(self):
     return self.__str__()
 
+  def __setattr__(self, attribute_name: str, attribute_value: Any):
```

Comment:
This is a pretty big side effect to have for any time we touch this object.  I can't remember offhand, do we instantiate this anywhere on the client still, or is that completely separate now?  I'm curious how many writes this will result in (and whether it slows anything down).  We might want to consider doing this explicitly via one of the other methods rather than implicitly here every time we touch the object.  If we do it this way we're going to have to be pretty careful to have good error handling in every single place that touches evidence objects otherwise we might start to see errors that are difficult to debug from the user perspective since the errors may only go in the logs and not get returned to the user.

Reply:
Let's chat about this offline. I think both approaches have pros and cons. The logic behind using __setattr__ is to aim for better consistency vs only updating in certain methods.